specification/storage/data-plane/Microsoft.QueueStorage/preview/2018-03-28/queue.json

```diff
@@ -1862,6 +1862,17 @@
       "maximum": 604800,
       "x-ms-parameter-location": "method",
       "description": "Optional. Specifies the new visibility timeout value, in seconds, relative to server time. The default value is 30 seconds. A specified value must be larger than or equal to 1 second, and cannot be larger than 7 days, or larger than 2 hours on REST protocol versions prior to version 2011-08-18. The visibility timeout of a message can be set to a value later than the expiry time."
+    },
+    "VisibilityTimeoutForEnqueue": {
+      "name": "visibilitytimeout",
+      "x-ms-client-name": "visibilityTimeout",
```

Comment:
This is using a different casing than previously, `visibility**T**imeout` vs `visibility**t**imeout`, is that going to cause an issue with generation?

Reply:
I first went without `"x-ms-client-name": "visibilityTimeout"` but it turned out to be named `visibilitytimeout` in JavaScript generated code, while before this change, the original name has camel casing - `visibilityTimeout`.

I don't want to change the name here, i.e. want to keep the camel casing as it was; so add the `x-ms-client-name` to enforce it.

Reply:
Turns out we have this transform in JavaScript README.md

### Rename visibilitytimeout -> visibilityTimeout

```yaml
directive:
  - from: swagger-document
    where: $.parameters.VisibilityTimeout
    transform: >
      $["x-ms-client-name"] = "visibilityTimeout";
  - from: swagger-document
    where: $.parameters.VisibilityTimeoutRequired
    transform: >
      $["x-ms-client-name"] = "visibilityTimeout";
```

Reply:
Hmm, looks like .NET don't have this transform.
https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/storage/Azure.Storage.Queues/swagger

I will remove `"x-ms-client-name": "visibilityTimeout"` in the swagger and add another transform in JavaScript.

Reply:
I think `"name": "visibilityTimeout"` is enough `x-ms-client-name` would be to override it.

Reply:
But to @alzimmermsft 's comment we should fix casing in `name`. Otherwise compilation may fail in .NET/Java

Reply:
Removed `"x-ms-client-name": "visibilityTimeout"` in https://github.com/Azure/azure-rest-api-specs/pull/11461